src/goto-programs/name_mangler.cpp

```diff
@@ -0,0 +1,48 @@
+/*******************************************************************\
+
+Module: Name mangling
+
+Author: Kareem Khazem <karkhaz@karkhaz.com>, 2019
+
+\*******************************************************************/
+
+#include "name_mangler.h"
+
+#include <iomanip>
+#include <sstream>
+
+irep_idt file_name_manglert::
+operator()(const symbolt &src, const std::string &extra_info)
+{
+  std::stringstream ss;
+  ss << CPROVER_PREFIX << "file_local_";
+  ss << std::regex_replace(
+          std::regex_replace(src.location.get_file().c_str(), forbidden, "_"),
+          multi_under,
+          "_")
+     << "_";
+
+  if(extra_info != "")
+    ss << extra_info << "_";
+  ss << src.name;
+  return irep_idt(ss.str());
+}
+
+irep_idt DJB_manglert::
+operator()(const symbolt &src, const std::string &extra_info)
+{
+  char const *str = src.location.get_working_directory().c_str();
+  unsigned long hash = 5381;
+  int c;
+  while((c = *str++))
+    hash = ((hash << 5) + hash) + c;
+
+  // Make no more than 8 nibbles wide
+  while(hash > 0xffffffffl)
```

Comment:
On a 32-bit UL arch this will surely generate a warning at least; suggest guarding by checking `sizeof(unsigned long)`

Reply:
(Or just hard cast to `uint32_t`? Is there any particular value in keeping the high rather than the low bits of the computed hash?)